be/src/storage/tablet_schema.h

```diff
@@ -229,6 +246,81 @@ class TabletColumn {
 bool operator==(const TabletColumn& a, const TabletColumn& b);
 bool operator!=(const TabletColumn& a, const TabletColumn& b);
 
+class TabletSchema;
+
+class TabletIndex {
+public:
+    TabletIndex() = default;
+    TabletIndex(const TabletIndex& other) noexcept {
+        TabletIndexPB index_pb;
+        other.to_schema_pb(&index_pb);
+        init_from_pb(index_pb);
+    }
+
+    void init_from_thrift(const starrocks::TOlapTableIndex& index, const TabletSchema& tablet_schema);
+    void init_from_pb(const TabletIndexPB& index);
+    void to_schema_pb(TabletIndexPB* index) const;
+    const std::string properties_to_json() const;
+
+    const int64_t index_id() const { return _index_id; }
+    const std::string& index_name() const { return _index_name; }
+    const IndexType index_type() const { return _index_type; }
+    const std::vector<int32_t>& col_unique_ids() const { return _col_unique_ids; }
+    const bool contains_column(int32_t column_uid) const {
+        return !_col_unique_ids.empty() && _col_unique_ids[0] == column_uid;
+    }
+
+    const std::map<std::string, std::string>& common_properties() const { return _common_properties; }
+    const std::map<std::string, std::string>& index_properties() const { return _index_properties; }
+    const std::map<std::string, std::string>& search_properties() const { return _search_properties; }
+    const std::map<std::string, std::string>& extra_properties() const { return _extra_properties; }
+
+    // ================ for ut ================
+    void add_common_properties(const std::string& key, const std::string& value) {
+        _common_properties.insert(std::make_pair(key, value));
+    }
+    void add_index_properties(const std::string& key, const std::string& value) {
+        _index_properties.insert(std::make_pair(key, value));
+    }
+    void add_search_properties(const std::string& key, const std::string& value) {
+        _search_properties.insert(std::make_pair(key, value));
+    }
+    void add_extra_properties(const std::string& key, const std::string& value) {
+        _extra_properties.insert(std::make_pair(key, value));
+    }
+
+    int64_t mem_usage() const {
+        int64_t mem_usage = sizeof(TabletIndex);
+        for (const auto& p : _common_properties) {
+            mem_usage += sizeof(char) * (p.first.size() + p.second.size());
+        }
+        for (const auto& p : _index_properties) {
+            mem_usage += sizeof(char) * (p.first.size() + p.second.size());
+        }
+        for (const auto& p : _search_properties) {
+            mem_usage += sizeof(char) * (p.first.size() + p.second.size());
+        }
+        for (const auto& p : _extra_properties) {
+            mem_usage += sizeof(char) * (p.first.size() + p.second.size());
+        }
+        mem_usage += sizeof(int32_t) * _col_unique_ids.size();
+        mem_usage += sizeof(char) * _index_name.size();
+
+        return mem_usage;
+    }
+
+private:
+    int64_t _index_id;
```

Comment:
```suggestion
    int64_t _index_id = -1;
```
Needs a default value.
I think int is enough for index id, no need to use int64_t